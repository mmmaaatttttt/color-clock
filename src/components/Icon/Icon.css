.Icon {
  color: #2196f3;
  background-color: white;
  border: 2px solid #2196f3;
  padding: 0.7rem;
  margin: 0 2rem;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: all 0.5s;
}

.Icon svg {
  width: 1.25em;
  height: 1.25em;
}

.Icon:hover {
  color: #606c76;
  border-color: #606c76;
  cursor: pointer;
}