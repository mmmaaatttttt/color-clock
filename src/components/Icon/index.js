import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import "./Icon.css";

function Icon({ onClick, faIcon }) {
  return (
    <div className="Icon" onClick={onClick}>
      <FontAwesomeIcon icon={faIcon} fixedWidth />
    </div>
  );
}

export default Icon;
