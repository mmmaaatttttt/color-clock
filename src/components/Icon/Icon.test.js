import React from "react";
import { render } from "@testing-library/react";
import Icon from "./";
import { faRedo, faEye } from "@fortawesome/free-solid-svg-icons";

describe("Icon component", function () {
  describe("basic tests", function () {
    it("renders without crashing", function () {
      render(<Icon faIcon={faRedo} />);
    });

    it("matches snapshot", function () {
      const { asFragment } = render(<Icon faIcon={faRedo} />);
      expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with different icon", function () {
      const { asFragment } = render(<Icon faIcon={faEye} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
