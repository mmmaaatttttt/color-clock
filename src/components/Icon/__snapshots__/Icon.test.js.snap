// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component basic tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Icon"
    role="button"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-redo fa-w-16 fa-fw "
      data-icon="redo"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M500.33 0h-47.41a12 12 0 0 0-12 12.57l4 82.76A247.42 247.42 0 0 0 256 8C119.34 8 7.9 119.53 8 256.19 8.1 393.07 119.1 504 256 504a247.1 247.1 0 0 0 166.18-63.91 12 12 0 0 0 .48-17.43l-34-34a12 12 0 0 0-16.38-.55A176 176 0 1 1 402.1 157.8l-101.53-4.87a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12h200.33a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12z"
        fill="currentColor"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`Icon component basic tests matches snapshot with different icon 1`] = `
<DocumentFragment>
  <div
    class="Icon"
    role="button"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-eye fa-w-18 fa-fw "
      data-icon="eye"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
        fill="currentColor"
      />
    </svg>
  </div>
</DocumentFragment>
`;
